pizjuce/MiddyMorphy/ControllerGUI.cpp

```diff
@@ -22,20 +21,20 @@ ControllerGUI::ControllerGUI (Controller* controller, MidiMorph* core)
     ccNo->setRange (0, 127, 1, 2);
     ccNo->addListener (this);
 
-    addAndMakeVisible (this->name = new TextEditor());
+    addAndMakeVisible (this->name = new juce::TextEditor());
     name->setSelectAllWhenFocused (true);
-    name->setColour (TextEditor::outlineColourId, Colour (0.f, 0, 0, 0.f));
-    name->setColour (TextEditor::backgroundColourId, Colour (0.f, 0, 0, 0.f));
-    name->setColour (TextEditor::focusedOutlineColourId, Colour (0.f, 0, 0, 0.f));
-    name->setColour (TextEditor::highlightColourId, Colours::white);
-    name->setFont (Font (10));
+    name->setColour (juce::TextEditor::outlineColourId, juce::Colour (0.f, 0, 0, 0.f));
+    name->setColour (juce::TextEditor::backgroundColourId, juce::Colour (0.f, 0, 0, 0.f));
+    name->setColour (juce::TextEditor::focusedOutlineColourId, juce::Colour (0.f, 0, 0, 0.f));
+    name->setColour (juce::TextEditor::highlightColourId, juce::Colours::white);
+    name->setFont (juce::Font (10));
     name->setInputRestrictions (32);
     name->addListener (this);
 
-    addAndMakeVisible (labCc = new Label ("", "cc:"));
-    addAndMakeVisible (labCh = new Label ("", "ch:"));
-    labCc->setFont (Font (10));
-    labCh->setFont (Font (10));
+    addAndMakeVisible (labCc = new juce::Label ("", "cc:"));
+    addAndMakeVisible (labCh = new juce::Label ("", "ch:"));
+    labCc->setFont (juce::Font (10));
+    labCh->setFont (juce::Font (10));
```

Comment:
I'm not perfectly sure about this as I think it helps a lot to be able to easily differentiate between e.g. project based components and components included from juce. It's simpler to grasp in which way something is composed and how many external elements are used.
I also don't like about it, that you still could write the prefixed version even when the namespace is included (as seen in this repo a lot). With my proposal it would be more consistent through out the whole code base and all external stuff would be prefixed. For me this beats the verbosity of it.

Reply:
I see your point. Still, there are some parts of the diff that are significantly harder for me to grasp now (as in: I need significantly more time to parse these lines).

The worst offenders to me are:
  - Math utility functions like `roundToInt`, `jmin`, `jmax`: These are usually part of expressions that contain a number of parentheses and casts and other operators. Getting to the actual calculation is made even harder when juggling namespaces at the same time.
  - Basic container data types, like `String`, `Array` or also `MidiMessage. At the places where they are used, it is perfectly clear, what these types represent.

The common thing here is that the role that these pieces fill in a given context is really obvious and most of the time, it's completely irrelevant, how or where they are implemented. On the other hand, if I want to find out, that is only one keypress away. As far as I am concerned, this is a case where an abstraction works really well. Strongly emphasizing that these ubiquitous APIs are part of JUCE then is forcing mostly irrelevant details onto the reader of the code. Conceptually, you can think of these as if they were our own helper functions/types whose implementations just happen to be really simple.

---

For the more specialized types (like `juce::ListBox` or `juce::NotificationType`), I can see a bit more usefulness in specifying that these are part of JUCE. So considering that it's likely that you will spend more time with the MiddyMorphy code than me, that would be fine for me.

Reply:
Thanks, I'll think about it. Time to sleep for now.
But I think we definitely shouldn't make the decision based on my possible future contributions.

I was under the impression that it's considered bad practice to use it (Google C++ Styleguide, Audioprogrammer Discord, Stackoverflow). But I might have misunderstood that.

Reply:
Alright, let's go with that for now. If excessive repetitions of `juce::` should bug me really strongly down the road, I will change it then.